recipes/arb-bio/6.1.rev16223/build.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+set -x
+
+#### "./configure" ####
+
+export ARBHOME=`pwd`
+export PATH="$ARBHOME/bin:$PATH"
+export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$ARBHOME/lib"
+
+export PKG_CONFIG_LIBDIR="$PREFIX/lib/pkgconfig:$PREFIX/share/pkgconfig"
+export XLIBS=$(pkg-config --libs xpm xerces-c)
+export XAW_LIBS=$(pkg-config --libs xaw7)
+export XML_INCLUDES=$(pkg-config --cflags xerces-c)
+export XINCLUDES=$(pkg-config --cflags x11)
+
+## Suppress building tools bundled with ARB for which we have
+# conda packages:
+export ARB_BUILD_SKIP_PKGS="MAFFT MUSCLE RAxML PHYLIP FASTTREE MrBAYES"
+
+# ARB stores build settings in config.makefile. Create one from template:
+cp config.makefile.template config.makefile
+
+# Now add some target specific settings to config.makefile
+case `uname` in
+    Linux)
+	echo DARWIN := 0
+	echo LINUX := 1
+	echo MACH := LINUX
+	echo LINK_STATIC := 0
+	SHARED_LIB_SUFFIX=so
+	;;
+    Darwin)
+	echo DARWIN := 1
+	echo LINUX := 0
+	echo MACH := DARWIN
+	echo LINK_STATIC := 0
+	SHARED_LIB_SUFFIX=dylib
+	export CFLAGS="$CFLAGS -w"
+	# needed for ARB Perl Binding compilation using MakeMaker
+	export LDFLAGS="$LDFLAGS -Wl,-rpath,$PREFIX/lib"
+	;;
+esac >> config.makefile
+
+#### "make" ####
+
+make SHARED_LIB_SUFFIX=$SHARED_LIB_SUFFIX -j$CPU_COUNT build | sed 's|'$PREFIX'|$PREFIX|g'
+
+#### "make install" ####
+
+# create tarballs (picks the necessary files out of build tree)
+make SHARED_LIB_SUFFIX=$SHARED_LIB_SUFFIX tarfile_quick | sed 's|'$PREFIX'|$PREFIX|g'
+
+# unpack tarballs at $PREFIX
+ARB_INST=$PREFIX/lib/arb
+mkdir $ARB_INST
+tar -C $ARB_INST -xzf arb.tgz
+tar -C $ARB_INST -xzf arb-dev.tgz
+
+# symlink arb_* executables into PATH
+(
+ cd $PREFIX/bin;
+ for exe in $ARB_INST/bin/arb*; do
+     ln -s "$exe"
+ done
+)
+
+# fix the library paths
+case `uname` in
+    Darwin)
+	# fix library IDs
```

Comment:
"Normally", this would be done by libtool. Under Linux, ARB uses LD_LIBRARY_PATH liberally to force it to work. Under OSX, that's not possible since Apple introduced that annoying SIP thing that removes linker settings from the environment. Macports side steps the issue by linking statically. I've tried a lot of approaches to get this to be right at build time, without post-build-tinkering, but this has been the simplest and most robust way to get the paths to be understood by conda's own path-fixing.